packages/block-library/src/columns/editor.scss

```diff
@@ -89,30 +89,24 @@
 
 			// Beyond mobile, allow 2 columns.
 			@include break-small() {
-				flex-basis: 50%;
+				flex-basis: calc(50% - #{$grid-size-large});
```

Comment:
Need to account for the additional padding added in the editor.

`flex-basis: calc(50% - #{$grid-size-large} - #{$block-padding * 2});`

Reply:
As noted in feedback here, https://github.com/WordPress/gutenberg/pull/12816/files#r242051845, this has an unintentional side-effect. Are you _sure_ it needs this additional padding into consideration? I believe it does not, because of the negative margins present in the editor. Yeah this is a bit of a headache, I'd love refactoring this in the future to simplify things, but for now I think we might need separate rules for the editor and the front-end.